lms/views/api/blackboard/files.py

```diff
@@ -20,28 +20,40 @@ def __init__(self, request):
         self.blackboard_api_client = request.find_service(name="blackboard_api_client")
 
     @view_config(request_method="GET", route_name="blackboard_api.courses.files.list")
+    @view_config(
+        request_method="GET", route_name="blackboard_api.courses.files.list_folder"
+    )
     def list_files(self):
         """Return the list of files in the given course."""
 
-        files = self.blackboard_api_client.list_files(
-            self.request.matchdict["course_id"]
-        )
-
-        pdf_files = []
-
-        for file in files:
-            if file.get("mimeType") != "application/pdf":
-                continue
-
-            pdf_files.append(
-                {
-                    "id": f"blackboard://content-resource/{file['id']}/",
-                    "display_name": file["name"],
-                    "updated_at": file["modified"],
-                }
-            )
-
-        return pdf_files
+        course_id = self.request.matchdict["course_id"]
+        folder_id = self.request.matchdict.get("folder_id")
+
+        results = self.blackboard_api_client.list_files(course_id, folder_id)
+
+        response_results = []
+
+        for result in results:
+            response_result = {
+                "display_name": result["name"],
+                "updated_at": result["modified"],
+                "type": result["type"],
+                "parent_id": folder_id,
+            }
+
+            if result["type"] == "File" and result.get("mimeType") == "application/pdf":
+                response_result["id"] = f"blackboard://content-resource/{result['id']}/"
+                response_results.append(response_result)
+            elif result["type"] == "Folder":
+                response_result["id"] = result["id"]
+                response_result["contents"] = self.request.route_path(
+                    "blackboard_api.courses.files.list_folder",
+                    course_id=course_id,
+                    folder_id=result["id"],
+                )
```

Comment:
This should maybe be a dict with `authUrl` and `path` as suggested earlier

Reply:
Yeah, probably a good idea for consistency

Reply:
Ok, changed this to use the same dict format as used elsewhere: https://github.com/hypothesis/lms/pull/2950/commits/79f6617b842085c9c8ca045fc81d29840c21d78f